internal/command/diff_inputs.go

```diff
@@ -0,0 +1,390 @@
+	repo := mustFindRepository()
+
+	taskRun, err := getTaskRun(repo, app, task, runID)
+	if err != nil {
+		exitOnErr(err)
+	}
+
+	var inputs *baur.Inputs
+	if taskRun == nil {
+		task := mustArgToTask(repo, fmt.Sprintf("%s.%s", app, task))
+
+		inputResolver := baur.NewInputResolver()
+
+		inputFiles, err := inputResolver.Resolve(ctx, repo.Path, task)
+		if err != nil {
+			exitOnErr(err)
+		}
+
+		inputs = baur.NewInputs(baur.InputAddStrIfNotEmpty(inputFiles, c.inputStr))
+	} else {
+		psql := mustNewCompatibleStorage(repo)
+		storageInputs, err := psql.Inputs(ctx, taskRun.ID)
+
+		if err != nil {
+			exitOnErr(err)
+		}
+
+		// Convert the inputs from the DB into baur.Input interface implementation
+		var baurInputs []baur.Input
+		for _, input := range storageInputs {
+			baurInputs = append(baurInputs, &storageInput{input})
+		}
+
+		inputs = baur.NewInputs(baur.InputAddStrIfNotEmpty(baurInputs, c.inputStr))
+	}
+
+	return inputs, taskRun, nil
+}
+
+func getTaskRun(repo *baur.Repository, app, task, runID string) (*storage.TaskRunWithID, error) {
+	if runID == "" {
+		return nil, nil
+	}
+
+	if strings.Contains(runID, "^") {
+		return getPreviousTaskRun(repo, app, task, runID)
+	}
+
+	id, err := strconv.Atoi(runID)
+	if err != nil {
+		exitOnErr(err)
+	}
+	return getTaskRunByID(repo, id)
+}
+
+func getPreviousTaskRun(repo *baur.Repository, app, task string, position string) (*storage.TaskRunWithID, error) {
+	var filters []*storage.Filter
+
+	filters = append(filters, &storage.Filter{
+		Field:    storage.FieldApplicationName,
+		Operator: storage.OpEQ,
+		Value:    app,
+	})
+
+	filters = append(filters, &storage.Filter{
+		Field:    storage.FieldTaskName,
+		Operator: storage.OpEQ,
+		Value:    task,
+	})
+
+	sorters := getSorters()
+
+	psql := mustNewCompatibleStorage(repo)
+
+	var taskRuns []*storage.TaskRunWithID
+	err := psql.TaskRuns(
+		ctx,
+		filters,
+		sorters,
+		func(taskRun *storage.TaskRunWithID) error {
+			taskRuns = append(taskRuns, taskRun)
+			return nil
+		},
+	)
+
+	if err != nil {
+		exitOnErr(err)
+	}
+
+	if len(taskRuns) < len(position) {
+		exitOnErr(fmt.Errorf("%s.%s%s does not exist, only %d task-run(s) exist(s)", app, task, position, len(taskRuns)))
+	}
+
+	taskRun := taskRuns[len(position)-1]
+
+	return taskRun, nil
+}
+
+func getTaskRunByID(repo *baur.Repository, id int) (*storage.TaskRunWithID, error) {
+	var filters []*storage.Filter
+	filters = append(filters, &storage.Filter{
+		Field:    storage.FieldID,
+		Operator: storage.OpEQ,
+		Value:    id,
+	})
+
+	sorters := getSorters()
+
+	psql := mustNewCompatibleStorage(repo)
+
+	var taskRun *storage.TaskRunWithID
+	err := psql.TaskRuns(
+		ctx,
+		filters,
+		sorters,
+		func(run *storage.TaskRunWithID) error {
+			taskRun = run
+			return nil
+		},
+	)
+
+	if err != nil {
+		if err == storage.ErrNotExist {
+			exitOnErr(fmt.Errorf("task-run %d does not exist", id))
+		}
+		exitOnErr(err)
+	}
+
+	return taskRun, nil
+}
+
+func getSorters() []*storage.Sorter {
```

Comment:
this functions seems like it could be replace with a variable, it always returns a slice with the same content

```go
  var defSorters = []*storage.Sorter{
          storage.Sorter{
                  Field: storage.FieldID,
                  Order: storage.OrderDesc,
          },
  }
```